Tour-Collective-Devs/tc_backend
# Rm rename user model

## Link to Ticket

Autolink format: hashtag and number, ex: #1094

Closes #

## Description of Proposed Changes
this pr renames everything referencing the old employer model to now call it by user

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout rm-rename-user-model
```
1. completely nuke your old database and migrations
1. no really. make sure there are absolutely no folders named 'migrations' in your database jacob. This caused me another 10 minutes of debugging. 
1. remake migrations and migrate database
1. load up server and make sure everything works exactly as before
1. you should be able to register and new user at '../user/registration'
1. and login/logout that user at '../user/auth/login' '../user/auth/logout'

## Impacted Areas in Application
renamed everything to do with the old employer model

List general components of the application that this PR will affect:

## Mentions @username

Tag users that need to review this code

## Testing

[ ] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[ ] I certify that all existing tests pass

## Deploy Notes

Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.

## Definition of Done

1. The project must be fully documented. This includes the following:
1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
1. Every class must be documented with purpose, author, and methods.
1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
1. Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each
